evalml/tests/demo_tests/test_datasets.py

```diff
@@ -1,10 +1,23 @@
+import urllib.request
+
 import pandas as pd
+import pytest
 
 from evalml import demos
 
 
+@pytest.fixture(autouse=True)
+def set_testing_headers():
+    opener = urllib.request.build_opener()
+    opener.addheaders = [("Testing", "True")]
+    urllib.request.install_opener(opener)
+
+
 def test_fraud():
     X, y = demos.load_fraud()
```

Comment:
The purpose of keeping this line is to make sure the api requests work?

Reply:
Yep, and to catch any issues if for any reason the local and S3 versions ever get out of sync.

Reply:
So this unit test will fail if there's no network connection? Are there others which fail under that scenario?

Reply:
Yeah, these ones will fail if there's no network connection but they're the only ones that will. I can pull them out into a separate test file so it's clearer that it's a separate test. It might also be possible to conditionally xfail if the tests can't reach S3 so they still work offline without causing failures.

Reply:
@dsherry I split out checking that the local and downloaded versions match and marked them to be skipped if the update server is offline. The tests won't fail if you run them offline, and they should still catch if something's wrong with the individual S3 links. Let me know what you think?